src/commonMain/kotlin/de/dbaelz/konclik/KonclikDsl.kt

```diff
@@ -4,6 +4,8 @@ package de.dbaelz.konclik
 annotation class KonclikDsl
 
 fun konclikApp(block: KonclikAppBuilder.() -> Unit): KonclikApp = KonclikAppBuilder().apply(block).build()
+fun konclikApp(args: Array<String>, block: KonclikAppBuilder.() -> Unit): Unit =
```

Comment:
I'd like the idea of this addition, but I'm ambivalent about the current implementation. We end up with two functions with the same name and different behavior. The first returns an instance and the other one runs/executes the application.

My current favorite is to rename your method to something like `runKonclikApp` or `executeKonclikApp` just to make it clear. What do you think?


PS: Maybe `KonclikApp#run()` isn't the best method name after all. I first called the method `execute` and changed it  again.

Reply:
I originally named the method `runKonclikApp` but felt the distinction was superfluous, I think the default behavior the DSL serves should be to build and run.  I imagine the majority of consumers will be building and running a `KonclikApp` without needing the reference at all.   Otherwise prefixing with the `run` or `execute` makes sense to me.  I do prefer `KonclikApp#execute`.

Reply:
Yes, I get the argument. We don't have to complicate things unnecessarily. Just keep the current naming and see if it fits.